pkg/extensions/search/resolver.go

```diff
@@ -238,170 +251,330 @@ func (r *queryResolver) repoListWithNewestImage(ctx context.Context, store stora
+		} else if imageLastUpdated != nil && repoLastUpdatedTimestamp.After(*imageLastUpdated) {
+			repoLastUpdatedTimestamp = *imageLastUpdated
+			lastUpdatedImageSummary = &imageSummary
 		}
 
-		if index := strings.Index(repo, name); index != -1 {
-			repoSize := strconv.FormatInt(repoSize, 10)
+		repoDownloadCount += manifestMetaMap[manifestDigest].DownloadCount
+	}
+
+	// calculate repo size = sum all manifest, config and layer blobs sizes
+	for _, blobSize := range repoBlob2Size {
+		size += blobSize
+	}
+
+	repoSize := strconv.FormatInt(size, 10)
+	score := 0
+
+	repoPlatforms := make([]*gql_generated.OsArch, 0, len(repoPlatformsSet))
+	for _, osArch := range repoPlatformsSet {
+		repoPlatforms = append(repoPlatforms, osArch)
+	}
+
+	repoVendors := make([]*string, 0, len(repoVendorsSet))
+
+	for vendor := range repoVendorsSet {
+		vendor := vendor
+		repoVendors = append(repoVendors, &vendor)
+	}
+
+	return &gql_generated.RepoSummary{
+		Name:          &repoName,
+		LastUpdated:   &repoLastUpdatedTimestamp,
+		Size:          &repoSize,
+		Platforms:     repoPlatforms,
+		Vendors:       repoVendors,
+		Score:         &score,
+		NewestImage:   lastUpdatedImageSummary,
+		DownloadCount: &repoDownloadCount,
+		StarCount:     &repoStarCount,
+		IsBookmarked:  &isBookmarked,
+		IsStarred:     &isStarred,
+	}, nil
+}
 
-			repos = append(repos, &gql_generated.RepoSummary{
-				Name:        &repo,
-				LastUpdated: &lastUpdatedTag.Timestamp,
-				Size:        &repoSize,
-				Platforms:   repoPlatforms,
-				Vendors:     repoVendors,
-				Score:       &index,
-				NewestImage: &lastUpdatedImageSummary,
-			})
+func imageHasSignatures(signatures map[string][]string) bool {
+	//  (sigType, signatures)
+	for _, sigs := range signatures {
+		if len(sigs) > 0 {
+			return true
 		}
 	}
 
-	sort.Slice(repos, func(i, j int) bool {
-		return *repos[i].Score < *repos[j].Score
-	})
+	return false
+}
 
-	sort.Slice(images, func(i, j int) bool {
-		return *images[i].Score < *images[j].Score
-	})
+func searchingForRepos(query string) bool {
+	return !strings.Contains(query, ":")
+}
 
-	sort.Slice(layers, func(i, j int) bool {
-		return *layers[i].Score < *layers[j].Score
-	})
+// updateRepoBlobsMap adds all the image blobs and their respective size to the repo blobs map
+// and returnes the total size of the image.
+func updateRepoBlobsMap(manifestDigest string, manifestSize int64, configDigest string, configSize int64,
+	layers []ispec.Descriptor, repoBlob2Size map[string]int64,
+) int64 {
+	imgSize := int64(0)
+
+	// add config size
+	imgSize += configSize
+	repoBlob2Size[configDigest] = configSize
+
+	// add manifest size
+	imgSize += manifestSize
+	repoBlob2Size[manifestDigest] = manifestSize
+
+	// add layers size
+	for _, layer := range layers {
+		repoBlob2Size[layer.Digest.String()] = layer.Size
+		imgSize += layer.Size
+	}
+
+	return imgSize
+}
+
+func getImageLastUpdated(configContent ispec.Image) *time.Time {
+	var lastUpdated *time.Time
```

Comment:
Since it's an optional field I considered *nil* as a good default value. It tells us there is no info about the time when the image it was last updated

Reply:
ok